Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -437,7 +429,8 @@ def evalNatCast : PositivityExt where eval {u α} _zα _pα e := do
 
 @[positivity Int.cast _]
 def evalIntCast : PositivityExt where eval {u α} _zα _pα e := do
-  let (.app _ (a : Q(Int))) ← withReducible (whnf e) | throwError "not Int.cast"
+  let _iα : Q(IntCast $α) ← synthInstanceQ q(IntCast $α)
+  let ~q(Int.cast ($a : ℤ)) := e | throwError "not Int.cast"
```

Comment:
Arguably using `@` would be better here, as that saves a TC search.

Reply:
Done in https://github.com/leanprover-community/mathlib4/pull/8098/commits/3a2cfb36b1104db495ed4517267814f0f68a9f16

Reply:
Are we worried about `Int.cast` having the wrong `IntCast` instance? It looks like it incidentally was checking that the `Int.cast` expression had the canonical one.